src/community/schemaless-features/schemaless-mongo/src/main/java/org/geoserver/schemalessfeatures/mongodb/filter/MongoSchemalessHelper.java

```diff
@@ -0,0 +1,161 @@
+/* (c) 2021 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.schemalessfeatures.mongodb.filter;
+
+import com.mongodb.BasicDBList;
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBObject;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.bson.Document;
+import org.geoserver.schemalessfeatures.mongodb.MongoSchemalessUtils;
+import org.geotools.util.logging.Logging;
+import org.opengis.feature.type.Name;
+
+/**
+ * Class helping to overcome those issues caused by the schemaless approach, eg. the retrieval of
+ * the type binding for a PropertyName or of a geometry path for the geometryDescriptor.
+ */
+public class MongoSchemalessHelper {
```

Comment:
I called it "dateTypeFinder" because originally the class name was indeed MongoTypeFinder, I don't know exactly why at some point I decided that MongoSchemlessHelper sound better... Fixed.